serde-support/src/main/java/io/micronaut/serde/support/serdes/SpecificObjectDeserializerArraySerde.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.serde.support.serdes;
+
+import io.micronaut.core.type.Argument;
+import io.micronaut.serde.Decoder;
+import io.micronaut.serde.Deserializer;
+
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+/**
+ * Deserializer for object arrays.
+ *
+ * @author graemerocher
+ * @since 1.0.0
+ */
+public class SpecificObjectDeserializerArraySerde implements Deserializer<Object[]> {
```

Comment:
This is only a deserializer so name seems inappropriate. I Would go with `CustomizedObjectArrayDeserializer`

Reply:
Can this be made final?